framework/db/QueryTrait.php

```diff
@@ -363,14 +363,21 @@ public function addOrderBy($columns)
      */
     protected function normalizeOrderBy($columns)
     {
+        $result = [];
         if ($columns instanceof ExpressionInterface) {
             return [$columns];
         } elseif (is_array($columns)) {
-            return $columns;
+            foreach ($columns as $column => $value) {
+                if (preg_match('#(asc|desc)#i', $value, $matches)) {
```

Comment:
This is partial match as well (like in "aaaascbbb") - is it intended? Otherwise `#^(asc|desc)$#i`.

Reply:
change to `#\b(asc|desc)\b#i`
Work with space (typo) before and after